controls/V-73173.rb

```diff
@@ -1,81 +1,64 @@
+/etc/opasswd."
   desc  "
-    Without generating audit records that are specific to the security and mission
-needs of the organization, it would be difficult to establish, correlate, and
-investigate the events relating to an incident or identify those responsible for one.
+    Without generating audit records that are specific to the security and
+mission needs of the organization, it would be difficult to establish,
+correlate, and investigate the events relating to an incident or identify those
+responsible for one.
 
-    Audit records can be generated from various components within the information
-system (e.g., module or policy filter).
+    Audit records can be generated from various components within the
+information system (e.g., module or policy filter).
   "
+if file(@audit_file).exist?
   impact 0.5
-  tag "severity": "medium"
+else
+  impact 0.0
+end
   tag "gtitle": "SRG-OS-000004-GPOS-00004"
   tag "gid": "V-73173"
-  tag "rid": "SV-87825r2_rule"
+  tag "rid": "SV-87825r4_rule"
   tag "stig_id": "RHEL-07-030874"
-  tag "cci": "CCI-000018"
-  tag "nist": ["AC-2 (4)", "Rev_4"]
-  tag "cci": "CCI-000172"
-  tag "nist": ["AU-12 c", "Rev_4"]
-  tag "cci": "CCI-001403"
-  tag "nist": ["AC-2 (4)", "Rev_4"]
-  tag "cci": "CCI-002130"
-  tag "nist": ["AC-2 (4)", "Rev_4"]
+  tag "cci": ["CCI-000018", "CCI-000172", "CCI-001403", "CCI-002130"]
+  tag "documentable": false
+  tag "nist": ["AC-2 (4)", "AU-12 c", "AC-2 (4)", "AC-2 (4)", "Rev_4"]
   tag "subsystems": ['audit', 'auditd', 'audit_rule']
   tag "check": "Verify the operating system must generate audit records for all
 account creations, modifications, disabling, and termination events that affect
-/etc/opasswd.
+/etc/security/opasswd.
 
-Check the auditing rules in \"/etc/audit/rules.d/audit.rules\" with the following
-command:
+Check the auditing rules in \"/etc/audit/rules.d/audit.rules\" with the
+following command:
 
-# grep /etc/opasswd /etc/audit/rules.d/audit.rules
+# grep /etc/security/opasswd /etc/audit/rules.d/audit.rules
 
--w /etc/opasswd -p wa -k audit_rules_usergroup_modification
+-w /etc/security/opasswd -p wa -k identity
 
 If the command does not return a line, or the line is commented out, this is a
 finding."
-
   tag "fix": "Configure the operating system to generate audit records for all
 account creations, modifications, disabling, and termination events that affect
-/etc/opasswd.
+/etc/security/opasswd.
 
-Add or update the following file system rule in \"/etc/audit/rules.d/audit.rules\":
+Add or update the following file system rule in
+\"/etc/audit/rules.d/audit.rules\":
 
--w /etc/opasswd -p wa -k identity
+-w /etc/security/opasswd -p wa -k identity
 
 The audit daemon must be restarted for the changes to take effect."
+  tag "fix_id": "F-79619r5_fix"
+
+  FILE_NAME = '/etc/security/opasswd'
 
-  @audit_file = '/etc/opasswd'
+  @audit_file = inspec.command('find /etc -type f -name "#{FILE_NAME}"').stdout.strip
```

Comment:
Odd. I made that change. What error are you seeing on your side @trevor-vaughan ?

Reply:
Ok I recall what I was trying to do here. 

What I was going for but I think it got lost in a merge was that the $FILE_NAME would be 'opasswd' and depending on your RHEL release that could be in /etc or in in /etc/security from version to version to RHEL. My intention behind the find command was to generalize this for all 'files' that we wanted to audit by using find to ensure we didn't need to care if RHEL moved things around in the /etc tree.

Thoughts?

Reply:
@aaronlippold The problem with making it a pattern is that you might match things that don't necessarily apply and will cause false positives.

If this is going to be reused code, it should get dropped into `libraries` and become a callable function instead of having the repeated snippets. If made into a function, then the search results could technically be aggregated into a data structure so that we're not thrashing the filesystem with `find` commands (caching *should* help here).

My vote is for making it an array of paths that you walk down and check where everything that is valid has to pass. Be careful with this, we also need to be sure that this isn't the same file before we have people add it to the `auditd` watch list. The more rules you add, the more load you add to the system over time and the STIG rules are bad enough as it is.

This is part of what's irritating about checking systems but, in some cases, there's just not much to be done.

Reply:
@trevor-vaughan I opted to just go back to using a single constant string. I haven't seen any actual evidence that we need to try to find alternate locations for the files.